src/Archetypes.cs

```diff
@@ -147,13 +155,21 @@ public void AddComponent(StorageType type, Identity identity, object data)
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     public bool HasComponent(StorageType type, Identity identity)
     {
+        if (identity == Identity.None)
```

Comment:
Early out feels wierd here, current HEAD already checks for Identity.none - so this can be...

```cs
    [MethodImpl(MethodImplOptions.AggressiveInlining)]
    public bool HasComponent(StorageType type, Identity identity)
    {
        var meta = Meta[identity.Id];
        return meta.Identity != Identity.None 
               && meta.Identity == identity 
               && Tables[meta.TableId].Types.Contains(type);
    }
```